platform-intel.c

```diff
@@ -669,14 +724,251 @@ static int read_efi_variable(void *buffer, ssize_t buf_size,
 	return 0;
 }
 
+/**
+ * is_efi_guid_equal() - check if EFI guids are equal.
+ * @guid: EFI guid.
+ * @guid1: EFI guid to compare.
+ *
+ * Return: %true if guid are equal, %false otherwise.
+ */
+bool is_efi_guid_equal(struct efi_guid guid, struct efi_guid guid1)
+{
+	int index;
+
+	for (index = 0; index < GUID_LENGTH; index++) {
+		if (guid.b[index] != guid1.b[index])
+			return false;
+	}
+	return true;
+}
+
+/**
+ * acpi_is_requested_uefi_controller() - check if read UEFI table contains requested controller.
```

Comment:
we have to highlight in naming that we looking for one from the list (first match wins).

Passing array does not match the "controller" in the name of the function. Perhaps, you should move `for loop` up, and here just compare with the one passed:

```c
bool acpi_is_requested_uefi_controller(uefi_controller_t *uefi_controller, struct uefi_acpi_table table)
{
     if (strncmp(table.oem_table_id, uefi_controller->name,
		strlen(uefi_controller->name)) == 0)
        return false;
    return is_efi_guid_equal(table.identifier, uefi_controller->guid);
}
```